homeassistant/components/enphase_envoy/sensor.py

```diff
@@ -411,12 +503,38 @@ def __init__(
         """Initialize Envoy entity."""
         super().__init__(coordinator, description)
         self._attr_unique_id = f"{self.envoy_serial_num}_{description.key}"
+        ct_count = coordinator.envoy.ct_meter_count
+        ct_consumptionmeter = coordinator.envoy.consumption_meter_type
+        phase_count = coordinator.envoy.phase_count
+        # Display phase count and found CT meters in model
+        model = (
+            "Envoy "
+            # Show CT Phase mode if CT is found
+            + (
+                f", phase mode: {coordinator.envoy.phase_mode}"
+                if ct_consumptionmeter is not None
+                else ""
+            )
+            # show number of phases found in Model description
+            + f" , phases: {str(phase_count)}"
+            # Show production CT found if 1 CT and no consumption ct of if more CT found
+            + (
+                ", production CT"
+                if (ct_count == 1 and ct_consumptionmeter is None) or (ct_count > 1)
+                else ""
+            )
+            # Add type of consumption CT to model if one was found
+            + (f", {ct_consumptionmeter} CT" if ct_consumptionmeter is not None else "")
+            + (", no CT" if ct_count == 0 else "")
+        )
         self._attr_device_info = DeviceInfo(
             identifiers={(DOMAIN, self.envoy_serial_num)},
             manufacturer="Enphase",
-            model=coordinator.envoy.part_number or "Envoy",
+            model=model,
             name=coordinator.name,
             sw_version=str(coordinator.envoy.firmware),
+            hw_version=coordinator.envoy.part_number or "Unknown",
```

Comment:
Can part_number be None? In that case omit the `or "Unknown"`, this will just hide it in the UI (Like, "Unknown" doesn't add value, so might as well just hide it)

Reply:
That section came from original code, only change is to show it in hardware version rather then model and change Envoy to Unknown.

```py
            model=coordinator.envoy.part_number or "Envoy",
```

```py
            hw_version=coordinator.envoy.part_number or "Unknown",
```
So it probably made sense in old setup and can be removed here.

Reply:
Unknown removed